tiny_dnn/core/kernels/conv2d_op_internal.h

```diff
@@ -28,42 +38,40 @@ inline void conv2d_op_internal(const tensor_t &in_data,
          serial_size_t kh          = params.weight.height_;
          serial_size_t elem_stride = params.w_stride;
          serial_size_t line_stride = iw * params.h_stride;
+         // TODO(edgarriba): replace with  tensor accessors
          for (size_t sample = r.begin(); sample < r.end(); sample++) {
-           const vec_t &in = in_data[sample];
-           vec_t &a        = out_data[sample];
+           float_t *out_data_begin      = out_data.host_pointer(sample, 0);
+           const float_t *in_data_begin = in_data.host_pointer(sample, 0);
+           const float_t *weight_begin  = weights.host_pointer(0, 0);
            for (serial_size_t o = 0; o < od; o++) {
-             float_t *pa = &a[params.out.get_index(0, 0, o)];
+             // TODO(Randl): naming
+             auto pa = &out_data_begin[params.out.get_index(0, 0, o)];
              for (serial_size_t inc = 0; inc < id; inc++) {
                if (!params.tbl.is_connected(o, inc)) continue;
-               serial_size_t idx;
-               idx                = params.weight.get_index(0, 0, id * o + inc);
-               const float_t *pw  = &W[idx];
-               idx                = params.in_padded.get_index(0, 0, inc);
-               const float_t *pin = &in[idx];
-               float_t *pout      = pa;
+               auto pw =
+                 &weight_begin[params.weight.get_index(0, 0, id * o + inc)];
+               auto pin = &in_data_begin[params.in_padded.get_index(0, 0, inc)];
+               auto pout = pa;
                for (serial_size_t y = 0; y < oh; y++) {
-                 const float_t *pin_line = pin;
+                 auto pin_line = pin;
                  for (serial_size_t x = 0; x < ow; x++) {
-                   const float_t *pin_element = pin_line;
-                   const float_t *pw_element  = pw;
+                   auto pw_element = pw;
                    float_t sum{0};
                    // should be optimized for small kernel(3x3,5x5)
-                   for (serial_size_t wy = 0; wy < kh; wy++) {    // NOLINT
+                   for (serial_size_t wy = 0; wy < kh; wy++) {  // NOLINT
+                     auto pin_element = &pin_line[iw * wy];
                      for (serial_size_t wx = 0; wx < kw; wx++) {  // NOLINT
-                       sum += pw_element[wx] * pin_element[wx];
+                       sum += *(pw_element++) * *(pin_element++);
                      }
-                     pw_element += kw;
-                     pin_element += iw;
                    }
-                   pout[x] += sum;
-                   pin_line += elem_stride;
+                   *(pout++) += sum;
+                   pin_line = std::next(pin_line, elem_stride);
                  }
-                 pout += ow;
-                 pin += line_stride;
+                 pin = std::next(pin, line_stride);
```

Comment:
In original code,
```cpp
                 pout += ow;
                 pin += line_stride;
``
`pout` is advanced with `ow` variable which is `params.out.width_` but newer code increments the `pout` pointer variable inside loop.

I think the old way is better since x86/x64 processor instructions can have address displacements.